lib/mirage/mirage.mli

```diff
@@ -764,6 +764,104 @@ val default_argv : argv impl
 val no_argv : argv impl
 (** [no_argv] Disable command line parsing and set argv to [|""|]. *)
 
+(** {2 Mimic device} *)
+
+(** A {i mimic} device is a {i context} which contains few informations to let
+    the user to [connect]/{!Mimic.resolve}/allocate a new resource to
+    communicate with a peer. Depending on what the context contains, the
+    protocol used to communicate can be TCP/IP, TLS or SSH (or something else).
+
+    Few packages deliver a {i context} which allows the user to communicate to a
+    server via a {i certain} protocol. The user is able to {i merge} these
+    provided contexts to be able to communicate with a server with multiple
+    protocols. Then, the choice is made dynamically according to what the
+    context contains. *)
+
+type mimic
+
+val mimic : mimic typ
+(** A simple device which define a {i certain} Mimic's context. *)
+
+val mimic_merge : mimic impl -> mimic impl -> mimic impl
+(** [mimic_merge ctx0 ctx1] merges given contexts into one. *)
+
+(** {2 Git configuration} *)
+
+(** The user is able to initiate a connection with a Git repository. However, it
+    exists multiple ways to communicate with a Git repository:
+
+    - TCP/IP
+    - HTTP
+    - HTTP + TLS
+    - SSH
+
+    These devices define these composable ways and deliver a Mimic's context
+    which can be pass to Git or Irmin in order to be able to {i fetch} and
+    {i push} from/into a Git repository. *)
+
+val git_happy_eyeballs :
+  ?random:random impl ->
+  ?time:time impl ->
+  ?mclock:mclock impl ->
+  ?pclock:pclock impl ->
+  stackv4v6 impl ->
+  mimic impl
+(** [git_happy_eyeballs stackv4v6] provides a {!mimic} device which is {b only}
+    able to resolve domain-name. The user must {!mimic_merge} this device with
+    another which provides a context with a {i certain} protocol. *)
```

Comment:
I'm not sure I understand what "certain" means here? From the context I'm guessing it's a communication protocol that supports arbitrary messages (e.g. TCP, SSH?)

Reply:
When I say _certain_ protocol, I mean any protocols. A concrete example is, of course, SSH or TCP/IP or even HTTP but it can be something else.